x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -188,9 +186,10 @@ export const createLifecycleExecutor = (
     `Tracking ${allAlertIds.length} alerts (${newAlertIds.length} new, ${trackedAlertStates.length} previous)`
   );
 
-  const alertsDataMap: Record<string, Partial<ParsedTechnicalFields>> = {
-    ...currentAlerts,
```

Comment:
@weltenwort Running it locally results in this error Executing Alert default:apm.error_rate:4b260f80-0b1f-11ec-b833-d530903bd1e0 has resulted in Error: fields is not defined. I guess it has to do with this line of code I removed. currentAlerts contains the alerts that the different rule alert types will populate when calling alertWithLifecycle.

I have a few ideas I want to try. In case you can see why it might be throwing this error, give a shout.

Reply:
This `alertData?.fields[ALERT_WORKFLOW_STATUS]` fixed the issue